src/Concordium/Client/Runner.hs

```diff
@@ -815,6 +815,15 @@ getAccountCfgFromTxOpts baseCfg txOpts = do
       let filteredKeys = Map.mapWithKey (\c m -> Map.filterWithKey (\k _ -> case Map.lookup c chosenKeys of
                                                                         Nothing -> False
                                                                         Just keyList -> elem k keyList) m) newKeys
+      _ <- Map.traverseWithKey (\c keyIndices -> case Map.lookup c newKeys of
+                                              Nothing -> logFatal [ "No credential holder with index: " ++ (show c)]
+                                              Just credHolderMap -> do
+                                                                  let warnIfMissingKey keyIndex = case Map.lookup keyIndex credHolderMap of
+                                                                          Nothing -> logFatal [ "Key index: " ++ (show keyIndex) ++ " is undefined for credential holder: " ++ (show c)]
```

Comment:
Maybe the error message could be consistent with the one for missing credentials. For example
```
No key with index ... for credential holder ...
```